File: services/crm/internal/apps/crm/models/tag.go
package models

import "time"

type Tag struct {
	ID        string    `bson:"_id" json:"id"`
	CrmID     string    `bson:"crmID" json:"crmID"`
	CreatedBy string    `bson:"createdBy" json:"createdBy"`
	Status    string    `bson:"status" json:"status"`
	Tag       string    `bson:"tag" json:"tag"`
	CreatedAt time.Time `bson:"createdAt" json:"createdAt"`
	UpdatedAt time.Time `bson:"updatedAt" json:"updatedAt"`
}

File: frontend/app/composables/api/catalog/tag.ts
import { useApi } from '../useApi'

interface ApiResult<T> {
  success: boolean
  data?: T
  error?: string
}

export interface CatalogTag {
  id: string
  crm_id: string
  name: string
  created_at: string
  updated_at: string
}

export const useCatalogTagAPI = () => {
  const api = useApi()

  const CreateTag = async (name: string): Promise<ApiResult<{ data: CatalogTag }>> => {
    try {
      const response = await api.post('/catalog/tags', { name })
      return { success: true, data: response.data }
    } catch (error: any) {
      return { success: false, error: error?.response?.data?.response_message || 'Failed to create tag' }
    }
  }

  const ListTags = async (): Promise<ApiResult<{ data: CatalogTag[] }>> => {
    try {
      const response = await api.get('/catalog/tags')
      return { success: true, data: response.data }
    } catch (error: any) {
      return { success: false, error: error?.response?.data?.response_message || 'Failed to fetch tags' }
    }
  }

  const DeleteTag = async (tagId: string): Promise<ApiResult<null>> => {
    try {
      await api.delete(`/catalog/tags/${tagId}`)
      return { success: true }
    } catch (error: any) {
      return { success: false, error: error?.response?.data?.response_message || 'Failed to delete tag' }
    }
  }

  return { CreateTag, ListTags, DeleteTag }
}
